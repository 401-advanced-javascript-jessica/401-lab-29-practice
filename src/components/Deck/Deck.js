import React from 'react';

const Deck = (props) => {
console.log(props);
    let deck = props.deck;
    return (
        <>
            <section className="deck">
                <header><h2>{deck.title}</h2></header>
                {props.children}
            </section>
        </>
    );
};

export default Deck;