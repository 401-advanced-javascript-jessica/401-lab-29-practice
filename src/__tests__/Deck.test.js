import React from 'react';
import renderer from 'react-test-renderer';
import Deck from '../components/Deck/Deck';


describe('<Deck />', () => {
  const deck = {
    title: 'My Deck of Stuff',
    records: [
      {
        title: 'Content Record 1',
        copy: 'Content Record 1 Copy Here...',
        media: {
          alt: 'This is a picture',
          title: 'This is the title of the picture',
          href: 'https://placehold.it/200x100',
        },
        links: [
          {
            type: 'primary',
            title: 'Click Me',
            href: 'https://www.foo.com',
          },
          {
            type: 'secodary',
            title: 'Not Me',
            href: 'https://www.bar.com',
          },
        ],
      },
      {
        title: 'Content Record 2',
        copy: 'Content Record 2 Copy Here...',
        media: {
          alt: 'This is a picture',
          title: 'This is the title of the picture',
          href: 'https://placehold.it/200x100',
        },
        links: [
          {
            type: 'primary',
            title: 'Click Me',
            href: 'https://www.foo.com',
          },
          {
            type: 'secodary',
            title: 'Not Me',
            href: 'https://www.bar.com',
          },
        ],
      },
      {
        title: 'Content Record 3',
        copy: 'Content Record 3 Copy Here...',
        media: {
          alt: 'This is a picture',
          title: 'This is the title of the picture',
          href: 'https://placehold.it/200x100',
        },
        links: [
          {
            type: 'primary',
            title: 'Click Me',
            href: 'https://www.foo.com',
          },
          {
            type: 'secodary',
            title: 'Not Me',
            href: 'https://www.bar.com',
          },
        ],
      },
    ],
  };

  it('Deck is rendered at application start', () => {
    const app = shallow(<Deck deck={deck}/>);
    expect(app.find('.deck').exists()).toBe(true);
  });

  it('content is rendered', () => {
    const app = mount(<Deck deck={deck}/>);
    expect(app.find('h2').text()).toEqual('My Deck of Stuff');
  });

  it('renders properly', () => {
    const tree = renderer.create(<Deck deck={deck}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
