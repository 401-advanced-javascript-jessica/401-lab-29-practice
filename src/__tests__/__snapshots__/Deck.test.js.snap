// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deck /> renders properly 1`] = `
<section
  className="deck"
>
  <header>
    <h2>
      My Deck of Stuff
    </h2>
  </header>
</section>
`;
